packages/core/src/VDocument.ts

```diff
@@ -156,4 +157,57 @@ export class VDocument {
             }
         }
     }
+    /**
+     * Insert a list at range.
+     *
+     * @param type
+     */
+    list(type: 'ordered' | 'unordered'): void {
+        let selectedBlocks: Set<VNode>;
+        if (this.range.isCollapsed()) {
+            selectedBlocks = new Set([this.range.start.parent]);
+        } else {
+            selectedBlocks = new Set(
+                this.range.selectedNodes.map(node => {
+                    return node.atomic ? node.parent : node;
```

Comment:
I'm not sure I understand why this distinction must be made.

Reply:
It's wrong. This doesn't handle nested blocks within list items. I'll rewrite it.